src/main/java/uk/gov/hmcts/cp/cdk/batch/tasklet/UploadAndPersistTasklet.java

```diff
@@ -45,145 +54,152 @@ public class UploadAndPersistTasklet implements Tasklet {
-                log.warn("Retrying getBlobSize (attempt #{}) path={}",
-                        ctx.getRetryCount() + 1, fDestBlobPath);
+                log.warn("jobId={} Retrying getBlobSize attempt #{} path={}", jobId, ctx.getRetryCount() + 1, destBlobPath);
             }
-            return storageService.getBlobSize(fDestBlobPath);
+            return storageService.getBlobSize(destBlobPath);
         }, (RetryContext ctx) -> {
-            log.warn("getBlobSize permanently failed after {} attempts. path={}",
-                    ctx.getRetryCount(), fDestBlobPath, ctx.getLastThrowable());
+            log.warn("jobId={} getBlobSize failed after {} attempts. path={}, cause={}", jobId, ctx.getRetryCount(), destBlobPath,
+                    ctx.getLastThrowable() != null ? ctx.getLastThrowable().getMessage() : "n/a", ctx.getLastThrowable());
             return null;
         });
-        final long sizeBytes = (sizeObj != null) ? sizeObj : -1L;
-
-        final CaseDocument caseDocumentEntity = new CaseDocument();
-        caseDocumentEntity.setDocId(persistedDocId);
-        caseDocumentEntity.setCaseId(caseId);
-        caseDocumentEntity.setMaterialId(materialId);
-        caseDocumentEntity.setDocName(blobName);
-        caseDocumentEntity.setBlobUri(blobUrl);
-        caseDocumentEntity.setContentType(contentType);
-        caseDocumentEntity.setSizeBytes(sizeBytes);
-        caseDocumentEntity.setUploadedAt(utcNow());
-        caseDocumentEntity.setIngestionPhase(DocumentIngestionPhase.UPLOADED);
-        caseDocumentEntity.setIngestionPhaseAt(utcNow());
-        caseDocumentRepository.saveAndFlush(caseDocumentEntity);
-
-        log.info("Saved CaseDocument: docId={}, caseId={}, sizeBytes={}", persistedDocId, caseId, sizeBytes);
-
-        return status;
+        final long sizeBytes = sizeObj != null ? sizeObj : -1L;
+
+        final CaseDocument entity = new CaseDocument();
+        entity.setDocId(docId);
+        entity.setCaseId(caseId);
+        entity.setMaterialId(materialId);
+        entity.setDocName(blobName);
+        entity.setBlobUri(blobUrl);
+        entity.setContentType(contentType);
+        entity.setSizeBytes(sizeBytes);
+        entity.setUploadedAt(utcNow());
+        entity.setIngestionPhase(DocumentIngestionPhase.UPLOADED);
+        entity.setIngestionPhaseAt(utcNow());
+
+        caseDocumentRepository.saveAndFlush(entity);
+
+        log.info("jobId={} Saved CaseDocument docId={}, caseId={}, materialId={}, sizeBytes={}, blobUri={}",
+                jobId, docId, caseId, materialId, sizeBytes, blobUrl);
+
+        stepCtx.put("uploaded_blob_path", destBlobPath);
+        stepCtx.put("uploaded_blob_url", blobUrl);
+        stepCtx.put("uploaded_size_bytes", sizeBytes);
+
+        return RepeatStatus.FINISHED;
+    }
+
+    @SuppressWarnings("PMD.OnlyOneReturn")
+    private UUID parseUuid(final ExecutionContext ctx, final String key, final String label, final String jobId) {
```

Comment:
we can do away with this if we were storing UUIDs.. lots of code can be removed.. please consider